### PR TITLE
Feature/ge post surveys

### LINKED ISSUE #2: Style Survey
**Feature:**
Before adding a monthly subscription I would like to choose my style preferences to ensure that I get clothes that better match my personal style. When I visit /crates, the first time I click on + subscribe I am taken to a style preferences page /style-preferences. I am presented with a style survey that consists of a series of categories such as tops, bottoms, dresses, shoes, accessories, etc. For each category I am given several images to choose from with instructions to choose the images that best reflects my style. After I have chosen the images from each category I am given a summary of my style such as ‘classic but edgy’.

**Extensions:**
* When I visit my profile page, I can see a button to view my style survey. When I click on the button, I am taken to /style-prefences and can edit my style preferences to update my style at any time.

* I would like to fill out a style survey for different categories of my wardrobe i.e. work, casual, date night. For each category I am presented with the same categories of choices i.e. tops, bottoms, dresses, shoes, accessories, etc.

### LINKED ISSUE #46: Create survey result functionality

### PR DESCRIPTION
**Pull request by GreysonElkins**
#### What's this PR do?
* Implements the feature where a user can post a survey to the server and their account is updated with their style
* Gives new users a role so they can also do surveys
* Allows admins to do surveys as well
#### Where should the reviewer start?
`Survey.js` and `Item.js` and `Survey/api/actions`
#### How should this be manually tested?
TAKE SURVEYS
#### Any background context you want to provide?
* We still don't know what happens when a user selects one of each style
* Sorry there's no testing - we just don't got that kinda time 
#### What are the relevant tickets?
#40, close #2, close #46, #39,  

### New Feature Submissions:
* [x] Have you added an explanation of what your changes do?
* [x] Have you linted your code locally prior to submission?
### Testing:
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?
* [ ] All new and existing tests passed?

### CODE REVIEW

**Comment by GreysonElkins** [comment]
**_we got it y'all_**
giphy

**Review by Joshua-Tukman — APPROVED** [review]
It was a pleasure to watch you frontenders get this last bit of frontend functionality working

File: code/web/src/modules/crate/Item.js [hunk 1]
```diff
@@ -120,6 +90,37 @@ class Item extends PureComponent {
...
+      owner.setState({
+        isLoading: false
+      })
+
+      window.setTimeout(() => {
+        owner.props.messageHide()
+      }, 5000)
+    })
+  }
+
```
Comment (NakiNorton) [inline code/web/src/modules/crate/Item.js:123]: Great idea to export this function so it's accessible to the Survey component.

File: code/api/src/modules/survey/resolvers.js [hunk 2]
```diff
@@ -27,15 +27,16 @@ export async function create(parentValue, { surveyContents }, { auth }) {
...
+    JSON.parse(surveyContents).forEach(question => {
+      if (styleCounter[question.answer]) {
+        styleCounter[question.answer].count += 1
+      } else {
+        styleCounter[question.answer] = { answer: question.answer, count: 1 }
+      }
+    })
+    const scoredResults = Object.values(styleCounter).sort((a, b) => b.count - a.count)
+
+    return printStyle(scoredResults)
```
Comment (NakiNorton) [inline code/api/src/modules/survey/resolvers.js:39]: Great work with this function, love the logic.

**Review by sciencefixion — APPROVED** [review]
Simply amazing

**Review by NakiNorton — APPROVED** [review]

**Review by michaeljevans — APPROVED** [review]
Looking good. Nice work all around.

**Pull request merged**